kafka/source/pkg/adapter/message.go

```diff
@@ -30,12 +30,17 @@ import (
 	"github.com/cloudevents/sdk-go/v2/binding"
 	"github.com/cloudevents/sdk-go/v2/extensions"
 	"github.com/cloudevents/sdk-go/v2/protocol/http"
+	"github.com/linkedin/goavro/v2"
 	"go.opencensus.io/trace"
 	"go.uber.org/zap"
 
 	sourcesv1beta1 "knative.dev/eventing-contrib/kafka/source/pkg/apis/sources/v1beta1"
 )
 
+type avroDecoder struct {
```

Comment:
Can we have this code in a separate file, or even better a separate module? like `adapter/datacodec/avro.go`?